vertx-web/src/main/java/io/vertx/ext/web/handler/impl/CorsHandlerImpl.java

```diff
@@ -164,7 +164,10 @@ public void handle(RoutingContext context) {
     HttpServerResponse response = context.response();
     String origin = context.request().headers().get(ORIGIN);
     if (origin == null) {
-      Utils.appendToMapIfAbsent(response.headers(), VARY, ",", ORIGIN);
+      // allowed origin: *
+      if (allowedOrigin == null && allowedOrigins == null) {
```

Comment:
when `allowedOrigin == null && allowedOrigins == null` it means that allow origin was configured with `*`? If so we should invert the condition no? We only want to add `Vary: Origin` when allow origin is not `*` right?

Reply:
I've updated the code and added helpers to make it more readable.